test/Tests/src/System/File_Spec.enso

```diff
@@ -419,22 +419,57 @@ spec =
             f.is_writable.should_be_false
 
             r2 = "FOO".write f on_existing_file=Existing_File_Behavior.Append
-            r2.should_fail_with File_Error.IO_Error
+            r2.should_fail_with File_Error.Access_Denied
             f.read . should_equal "foobar"
 
             r3 = "FOO".write f on_existing_file=Existing_File_Behavior.Overwrite
-            r3.should_fail_with File_Error.IO_Error
+            r3.should_fail_with File_Error.Access_Denied
             f.read . should_equal "foobar"
 
             r4 = "FOO".write f on_existing_file=Existing_File_Behavior.Backup
-            r4.should_fail_with File_Error.IO_Error
+            r4.should_fail_with File_Error.Access_Denied
             f.read . should_equal "foobar"
 
             f.is_writable.should_be_false
 
             set_writable f True
             f.delete
 
+        Test.specify "should fail if the parent directory does not exist" <|
+            parent = transient / "nonexistent"
+            parent.exists.should_be_false
+
+            f = parent / "foo.txt"
+            r1 = "txt".write f
+            r1.should_fail_with File_Error.Not_Found
+            # The write fails because the _parent_ directory did not exist.
+            r1.catch.file . should_equal parent
```

Comment:
I thought that it makes sense to actually report `Not_Found` with not the file itself, but the parent directory.

That is because - if the file itself does not exist - that is _not_ an error for File.write. The issue is that we cannot write the file if the parent directory (or its ancestor) does not exist - so IMO if the error contains the parent dir and not the file it expresses this more clearly.

@jdunkerley what do you think about this suggestion?

I know this is different from what Java does (throwing Not Found on the requested file, not parent dir), so I'm OK to revert this part. But IMO that way is clearer.

Reply:
This looks good to me.